### ISSUE TITLE
Optimizer

### ISSUE THREAD

**Issue by hdmGOAT** [hdmGOAT/trustformer#9]
- [ ] SGD / Adam
- [ ] Movement buffers
- [ ] Weight decay
- [ ] Learning rate schedule
